overlays/dprint.nix

```
@@ -0,0 +1,61 @@
+{ dprint
+, jq
+, fetchurl
+, writeText
+, runCommand
+, makeWrapper
+
+, dprintJson
+}:
+
+let
+  dprintConfig = builtins.fromJSON (builtins.readFile dprintJson);
+
+  hashes = {
+    "https://plugins.dprint.dev/typescript-0.85.0.wasm" = "sha256-FMMoXkWL7oTZYCbh108dJtziUKgvvGH6FwK/cglUGLo=";
+  };
+
+  prefetchPlugin = url: fetchurl {
+    inherit url;
+    sha256 = hashes."${url}" or (throw "No known hash for dprint plugin '${url}', please update release.nix");
+  };
+
+  plugins = map prefetchPlugin dprintConfig.plugins;
+
+  simplifiedConfig = builtins.toJSON {
+    inherit plugins;
+  };
+
+  # used to map nix paths back to their remote urls in the plugin cache
+  # manifest so that the original dprint.json can be used
+  urls = builtins.listToAttrs (map (p: {
+    name = "local:${builtins.unsafeDiscardStringContext p}";
+    value = "remote:${p.url}";
+  }) plugins);
+
+  simplifiedConfigFile = writeText "dprint.json" simplifiedConfig;
+
+  dprintCache = runCommand "dprint-cache" {
+    buildInputs = [ dprint jq ];
```

Technically these should be `nativeBuildInputs`, but I don't know if dprint's caches are platform independent and a use case for cross compilation is unlikely.

```suggestion
    nativeBuildInputs = [ dprint jq ];
```

---

The cached plugin is keyed for architecture at least so it definitely won't work for cross compilation.